### PR TITLE
image-rs: make tar reader async

### PR DESCRIPTION
**Pull request by Xynnn007**
before this commit, tar reader is sync, thus we need a separate os thread for every image layer pulling green thread. This would cost more memory and CPU resource because on linux a thread actually means another process.

This commit replace the tar crate to async version, thus we do not need another thread to handle this to decrease memory occupation.

Also, the original thread way would make the error of `unpack()` function missing because it would not be captured by original thread. Thus always a inclear `Send Error {...}` error message will be returned. The new way will not face such situation. cc @stevenhorsman 

Some abilities like permission maintain, ownership maintain are manually implemented as the original crate does not support well.

cc @arronwy

### CODE REVIEW

**Comment by Xynnn007** [comment]
Rebased and added another commit to replace `libc` with `nix`. This commit should be test on kata pipeline before merge.

**Comment by stevenhorsman** [comment]
> Rebased and added another commit to replace `libc` with `nix`. This commit should be test on kata pipeline before merge.

Sure - as I don't have any green ticks for the current image-rs bump in kata, I'll temporarily re-purpose it to test your branch

**Comment by stevenhorsman** [comment]
> > Rebased and added another commit to replace `libc` with `nix`. This commit should be test on kata pipeline before merge.
> 
> Sure - as I don't have any green ticks for the current image-rs bump in kata, I'll temporarily re-purpose it to test your branch

In https://github.com/kata-containers/kata-containers/actions/runs/9806138253/job/27078629312?pr=9828 the non-TEE, SEV and SNP tests passed and the TDX is failing with normal results, so I think this is okay from the kata integration tests. I will drop the change for now, but if this gets merged before the image-rs bump does then I'll update to pick this up too. Thanks!

**Review by fitzthum — APPROVED** [review]
LGTM although I am super familiar with the image-rs code.

**Comment by fitzthum** [comment]
@arronwy do these changes look good to you?

**Review by arronwy — APPROVED** [review]
Thanks @Xynnn007 LGTM!

**Comment by arronwy** [comment]
> @arronwy do these changes look good to you?

Yes, the primary concern for this pr is the possibility that crate `krata-tokio-tar` may not be actively maintained in future.

The init async support for tar is `async-tar` project https://github.com/dignifiedquire/async-tar  which is forked from `tar-rs`, and this project is not update for three years.

Next is `tokio-tar` https://github.com/vorot93/tokio-tar which is forked from `async-tar` and not update for one year and may PR is not review: https://github.com/vorot93/tokio-tar/pulls

Finally is the crate we use `krata-tokio-tar` https://github.com/edera-dev/tokio-tar which is forked from `tokio-tar` with several bug fixings.

After sync with @Xynnn007 if future have issues in this crate, we can fork and fix it.

@fitzthum any suggestions?

**Comment by fitzthum** [comment]
@arronwy @Xynnn007 

the state of those repos is far from ideal, but I guess we can make it work. Maybe we can get in touch with some of those maintainers and figure out what the status is. We can fork if we have to.

**Pull request merged**
